api/src/main/java/org/pmiops/workbench/db/dao/UserDao.java

```diff
@@ -48,7 +46,21 @@
 
   Set<DbUser> findByFirstRegistrationCompletionTimeNotNull();
 
-  default Map<Boolean, Long> getDisabledToCountMap() {
-    return DaoUtils.getAttributeToCountMap(findAll(), DbUser::getDisabled);
+  @Query(
+      "SELECT dataAccessLevel, disabled, CASE WHEN betaAccessBypassTime IS NOT NULL THEN TRUE ELSE FALSE END AS betaIsBypassed, COUNT(userId) AS userCount "
+          + "FROM DbUser "
+          + "GROUP BY dataAccessLevel, disabled, CASE WHEN betaAccessBypassTime IS NOT NULL THEN TRUE ELSE FALSE END "
+          + "ORDER BY NULL")
+  List<DataAccessLevelDisabledAndBetaBypassedToCountRow>
+      getDataAccessLevelDisabledAndBetaBypassedToCount();
```

Comment:
Can we write a test class for `UserDaoTest` and test all methods?

Reply:
Yep. This is my first Dao Test, and the only thing that's tricky is understanding when my autowired dao is cleared between tests.